### PR TITLE
MFT: new MC track task

### PR DESCRIPTION
**Pull request by diana0x0f**

### CODE REVIEW

**Comment by diana0x0f** [comment]
Hi @chiarazampolli,  I have added the new MFT MC task. Could you please check if everything is correct? Thanks!

**Comment by benedikt-voelkel** [comment]
@diana0x0f we will also see if the QC goes through since there is a small test that will automatically take your new QCs into account.
The CI will start at some point.

File: MC/config/QC/json/mft-tracks-mc.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      "tasks" : {
+        "TracksMC" : {
+          "active" : "true",
+          "className" : "o2::quality_control_modules::mft::QcMFTTrackMCTask",
+          "moduleName" : "QcMFT",
+          "detectorName" : "MFT",
+          "cycleDurationSeconds" : "30",
+          "maxNumberCycles" : "-1",
+          "dataSource_comment" : "The other type of dataSource is \"direct\", see basic-no-sampling.json.",
+          "dataSource" : {
```
Comment (JianLIUhep) [inline MC/config/QC/json/mft-tracks-mc.json:37]: If your sampling fraction is 100%, please use `direct` so that we can avoid using the dispatcher.

**Review by JianLIUhep — CHANGES_REQUESTED** [review]

File: MC/config/QC/json/mft-tracks-mc.json (hunk 1 above)
Comment (diana0x0f) [reply MC/config/QC/json/mft-tracks-mc.json:37]: Hi Jian, thanks, I have changed the dataSource type to `direct`.

**Comment by diana0x0f** [comment]
Hi @benedikt-voelkel ,
I tested the workflow locally without any problem, all the new QC objects were created. How can we reproduce the problem that occurred in the CI test? Thanks!

**Comment by benedikt-voelkel** [comment]
Hej @diana0x0f ,
the easiest would be under the following assumptions:
* your local development directory of `O2DPG` is located at `$O2DPG_DEV`.
* your working directory is clean (everything committed, nothing staged, no changed files)
* your development is in the commits on top, maybe it consists even of only one commit. But let's assume the last `${N}` commits are the ones that contain you development.

Then run
```bash
O2DPG_TEST_REPO_DIR=${O2DPG_DEV} O2DPG_TEST_HASH_BASE=HEAD~${N} ${O2DPG_ROOT}/test/run_workflow_tests.sh
```
That should run the same thing.

Let me know if you experience any issues.
More details can be found here: https://aliceo2group.github.io/simulation/docs/generators/generatorconfig.html#run-the-test-locally
(note that this specifically refers to `run_generator_tests.sh`, however you need `run_workflow_tests.sh` as stated above)

**Comment by benedikt-voelkel** [comment]
This needs https://github.com/AliceO2Group/QualityControl/pull/2115 which does not seem to be contained in any QC tag.
@Barthelemy are there any plans to include it and also to bump QC to a new tag in alidist which would then have the necessary developments?

**Comment by Barthelemy** [comment]
@benedikt-voelkel There will be a tag tomorrow. Is it ok to have a new tag or do you need a patch on an existing QC tag ?

**Comment by chiarazampolli** [comment]
If this is for MC anchored to data, we need to patch the QC tag that is used for MC for those data.

**Comment by Barthelemy** [comment]
Ok, so I am going to tag today a new version and if you want me to patch a specific tag, just let me know. It is very quick to do (perhaps ping me on MM).

**Comment by chiarazampolli** [comment]
Hello @diana0x0f , 
For which production would you like this PR to be included?
Chiara

**Comment by diana0x0f** [comment]
Hi @chiarazampolli , this new MC task was not targeted specifically for any particular production at the moment. Our intention was to incorporate it into the codebase for any upcoming central MC production.

**Review by benedikt-voelkel — APPROVED** [review]

**Comment by benedikt-voelkel** [comment]
@chiarazampolli I guess no labels needed, right?
From my side, good to be merged.

**Comment by benedikt-voelkel** [comment]
@TimoWilken 
I think the CI was successful on this already a few days ago, maybe I am mistaken. Now it is failing due to fetching RCT from CCDB.

**Comment by TimoWilken** [comment]
Hi @benedikt-voelkel, sorry, known issue: https://its.cern.ch/jira/browse/O2-4735. The next automatic rebuild should fix it. Let me know if it's urgent, then I'll restart it manually, or you can merge it right away.

**Comment by benedikt-voelkel** [comment]
Thanks!
Let's wait for the rebuild, then, I will merge.

**Comment by chiarazampolli** [comment]
Hello @benedikt-voelkel , 
Sorry for coming back late to this: if this is needed for PbPb apass3 and MC, then labels should be added.
Cheers,
Chiara

**Comment by benedikt-voelkel** [comment]
I understood that it is not urgent but as @diana0x0f said

> Hi @chiarazampolli , this new MC task was not targeted specifically for any particular production at the moment. Our intention was to incorporate it into the codebase for any upcoming central MC production.

No rush from my side.
But of course, it could be that @diana0x0f means to include it if possible before starting "any" new MCs. Could you comment?

Personally, I cannot comment.

**Comment by diana0x0f** [comment]
Hi @benedikt-voelkel , @chiarazampolli ,
we would like to have this included for the PbPb apass3 and MC that Chiara mentioned. 
What I meant with the previous comment was indeed to have it included for "any" new MCs.

**Comment by chiarazampolli** [comment]
Ciao @diana0x0f , 

I added your task here:

https://gitlab.cern.ch/bvolkel/o2dpgdocs/-/blob/main/docs/software/requests/2023pbpb_apass.md

Please, do like that next time you need a cherry-pick in a production tag.

Chiara

**Comment by chiarazampolli** [comment]
BTW, do we need also to update the QC? Otherwise your task might not work. If this is the case, we need to ask for a tag to @Barthelemy .

**Comment by diana0x0f** [comment]
Hi @chiarazampolli , thanks! This [task](https://github.com/AliceO2Group/QualityControl/pull/2115) in QC was included in [v1.135.0](https://github.com/AliceO2Group/QualityControl/releases/tag/v1.135.0).

**Comment by chiarazampolli** [comment]
Hello @diana0x0f , 

Then we need to retag QC. @Barthelemy , could you do so? We currently use v1.126.5 in production.

Note the PR by @lmassacr too: she might need an update as well for QC (https://github.com/AliceO2Group/O2DPG/pull/1531), and the one by @martenole (https://github.com/AliceO2Group/O2DPG/pull/1467). They also might need an update of QC. 

Chiara

**Comment by JianLIUhep** [comment]
Hi @chiarazampolli, @Barthelemy, 
I think the PR for MFT in QC was already retagged for [v1.126.6](https://github.com/AliceO2Group/QualityControl/releases/tag/v1.126.6), which also includes the fix for HMPID. Then, another patch to HMPID was released with [v1.126.7](https://github.com/AliceO2Group/QualityControl/releases/tag/v1.126.7). 

So, if I am not mistaken, now we might only need to retag QC to include QC updates for [#1531](https://github.com/AliceO2Group/O2DPG/pull/1531) and [#1467](https://github.com/AliceO2Group/O2DPG/pull/1467) if that's the case.

**Pull request merged**